rust/extractor/src/crate_graph.rs

```diff
@@ -178,20 +183,107 @@ fn emit_module_children(
         .collect()
 }
 
+fn emit_reexport(
+    crate_graph: &CrateGraph,
+    db: &dyn HirDatabase,
+    trap: &mut TrapFile,
+    uses: &mut HashMap<String, trap::Label<generated::Item>>,
+    import: ImportOrGlob,
+    name: &str,
+) {
+    let (use_, idx) = match import {
+        ImportOrGlob::Glob(import) => (import.use_, import.idx),
+        ImportOrGlob::Import(import) => (import.use_, import.idx),
+    };
+    let def_db = db.upcast();
+    let loc = use_.lookup(def_db);
+    let use_ = &loc.id.item_tree(def_db)[loc.id.value];
+
+    use_.use_tree.expand(|id, path, kind, alias| {
+        if id == idx {
+            let mut path_components = Vec::new();
+            match path.kind {
+                PathKind::Plain => (),
+                PathKind::Super(0) => path_components.push("self".to_owned()),
+                PathKind::Super(n) => {
+                    path_components.extend(std::iter::repeat_n("parent".to_owned(), n.into()));
+                }
+                PathKind::Crate => path_components.push("crate".to_owned()),
+                PathKind::Abs => path_components.push("".to_owned()),
+                PathKind::DollarCrate(_) => path_components.push("$crate".to_owned()),
```

Comment:
Can this case ever happen, aren't all macros fully expanded?

Reply:
I doubt it ever happens, just wanted to cover the case so we can look for it in the database.